File: streampipes-sdk/src/main/java/org/streampipes/sdk/utils/Datatypes.java
package org.streampipes.sdk.utils;

import org.streampipes.vocabulary.SO;
import org.streampipes.vocabulary.XSD;

import java.net.URI;

public enum Datatypes {

    Integer(XSD._integer),
    Long(XSD._long),
    Float(XSD._float),
    Boolean(XSD._boolean),
    String(XSD._string),
    Double(XSD._double),
    Number(URI.create(SO.Number)),
    Sequence(XSD._sequence);

    private URI uri;

    Datatypes(URI uri) {
        this.uri = uri;
    }

    public String toString() {
        return uri.toString();
    }
}

File: ui/src/app/dashboard-v2/sdk/model/datatypes.ts
import {Vocabulary} from "./vocabulary";

export class Datatypes {

    static readonly Integer = new Datatypes(Vocabulary.XSD, "integer");
    static readonly Long = new Datatypes(Vocabulary.XSD, "long");
    static readonly Float = new Datatypes(Vocabulary.XSD, "float");
    static readonly Boolean = new Datatypes(Vocabulary.XSD, "boolean");
    static readonly String = new Datatypes(Vocabulary.XSD, "string");
    static readonly Double = new Datatypes(Vocabulary.XSD, "double");
    static readonly Number = new Datatypes(Vocabulary.SO, "Number");

    private constructor(private readonly vocabulary: string, private readonly entity: string ) {

    }

    toUri() {
        return this.vocabulary + this.entity;
    }


}